src/components/Message/Message.js

```diff
@@ -6,17 +6,34 @@ import styles from './Message.scss'
 
 const messageColors = ['#ff6666', '#6d72c4', '#a37aa2', '#73886e', '#ef9820']
 
-function getMessageStyle() {
+function getMessageStyle(id) {
+  const color = messageColors[id % messageColors.length];
+  return { backgroundColor: color }
+}
+
+function toRad(x) {
+  return x * Math.PI / 180;
+}
+
+function haversineDistance(x1, y1, x2, y2) {
+  var R = 6371; // average earth radius in kilometers
+
+  var x1 = y2 - x2;
+  var dLat = toRad(x1);
+  var x2 = y1 - x1;
+  var dLon = toRad(x2)
```

Comment:
In JS, adding semicolon before newline is optional. I assume you knew this already but just wanted to say.. (or correct me if I'm wrong because I have not tried with ESLint and wonder what it would say for my coding style)